Feat/add wagmi

matthew-garrett opened the pull request:
- Adds wagmi
- Removes onboard

jeffywu commented on packages/shared/notionable-hooks/src/use-account.ts:
```diff
@@ -40,6 +40,14 @@ export function useAccountDefinition(network: Network | undefined) {
+    globalState: { wallet },
+  } = useNotionalContext();
+
+  return wallet;
```
Return a boolean here instead of an object

jeffywu commented on packages/shared/wallet/src/hooks/use-connect.ts:
```diff
@@ -1,89 +1,161 @@
+
+export const useWalletConnect = () => {
+  const { globalState, updateNotional } = useNotionalContext();
+  const wagmiAccount = useAccount();
```
Can you rename this import to `useWagmiAccount` so it's not mistaken for a notional account

jeffywu commented on packages/shared/wallet/src/hooks/use-connect.ts:
```diff
@@ -1,89 +1,161 @@
+import WalletConnect from '../images/wallet-connect.svg';
+import CoinbaseWallet from '../images/coinbase-wallet.svg';
+import Safe from '../images/safe.svg';
+import { Connector, useConnect, useDisconnect } from 'wagmi';
```
Can you rename the useConnect and useDisconnect imports to useWagmiConnect and useWagmiDisconnect

jeffywu commented on packages/shared/wallet/src/hooks/use-connect.ts:
```diff
@@ -1,89 +1,161 @@
+  'Coinbase Wallet': CoinbaseWallet,
+};
+
+export const useWalletConnect = () => {
```
Can we put this back to `useConnect` to distinguish from the WalletConnect product

jeffywu commented on packages/shared/wallet/src/connect-wallet-side-drawer/connect-wallet-side-drawer.tsx:
```diff
@@ -2,55 +2,47 @@ import { useCallback } from 'react';
-  const connected = useAccountReady(network);
-  const { clearWalletSideDrawer } = useSideDrawerManager();
+  const { connectWallet, displayConnectors } = useWalletConnect();
+  const { address } = useAccount();
```
I'd prefer not to use any `wagmi` functions outside of the `use-connect` hook. Can we get the address from global state instead?

jeffywu commented on packages/shared/wallet/src/settings-side-drawer/settings-side-drawer.tsx:
```diff
@@ -51,11 +47,7 @@ export const SettingsSideDrawer = ({
-      currentAddress: null,
-    });
-    window.location.reload();
+    // window.location.reload();
```
Can you remove the commented lines?

matthew-garrett committed:
build: adds most wagmi functionality

build: updates use change network

build: fixes wagmi disconnect race condition

build: fixes wagmi disconnect race condition

build: adds ignore

build: removes on board

build: clean up use connect

build: clean up

build: addresses comments

build: remove hardhat

build: fixes conflicts
